src/Chart/Chart.jsx

```diff
@@ -96,10 +102,31 @@ export default (ChartElement) => class ChartComponent extends Component {
         dataset.data.push(dataObject[dataset.dataProperty])
       }
     }
+    
+    data.datasets = this.populateWithColors(data.datasets)
 
     return data
   }
   
+  populateWithColors (datasets) {
+    // update the colors on the datasets.
+    const datasetColors = this.props.colorPalette(datasets.length, this.props.colorScale)
+    
+    // Set the colors of the dataset if needed by ChartElement.
+    datasetColors.datasetColors = datasetColors
+    
+    // Make a copy of each of the datasets before passing them along
+    datasets = datasets.map((dataset, index) => {
+      return {
+        ...dataset,
+        backgroundColor: dataset.backgroundColor || chroma(datasetColors[index]).alpha(0.4).css(),
```

Comment:
I will add in those two consts as they make it look cleaner. The only reason I'm checking to see if there is already a backgroundColor on the dataset, is this allows the user to set a specific color on the dataset if they choose.